###  Overview  
Exploratory analysis of wikipedia-movies.json which contains data obtained from wikipedia.org of movies from 1990 to 2018.

In [1]:
#Impor dependencies
import json
import pandas as pd
import numpy as np

In [12]:
#Create directory that links to our json file
file_dir = "C:\\Users\Singh Lab\Desktop\DataAnalyticsCourse\Module_8\course_work\movies_etl\\"

Now we can read the json file (DO NOT IMPORT AS A DATAFRAME!!) and do some cleanup. Then we can convert the cleaned up data into a dataframe.

In [13]:
#Load the json file
with open(f'{file_dir}wikipedia-movies.json', 'r') as file:
    wiki_movies_raw = json.load(file)

In [14]:
#Lets check how many records are in the file:
len(wiki_movies_raw)

7311

It is a good idea to stop right here and decide whether that count seems excessively large or small which could indicate a problem with the data. To be honest, 7311 seems like a small number considering the number of movies made over  28 year period

In [15]:
#Lets look at the dictionary structure and records:
wiki_movies_raw[5]

{'url': 'https://en.wikipedia.org/wiki/The_Ambulance',
 'year': 1990,
 'imdb_link': 'https://www.imdb.com/title/tt0099026/',
 'title': 'The Ambulance',
 'Directed by': 'Larry Cohen',
 'Produced by': ['Larry Cohen', 'Moctesuma Esparza', 'Robert Katz'],
 'Written by': 'Larry Cohen',
 'Starring': ['Eric Roberts',
  'James Earl Jones',
  'Red Buttons',
  'Megan Gallagher'],
 'Music by': 'Jay Chattaway',
 'Cinematography': 'Jacques Haitkin',
 'Edited by': ['Claudia Finkle', 'Armond Leibowitz'],
 'Productioncompany ': 'Epic Productions',
 'Distributed by': 'Triumph Releasing Corporation',
 'Release date': ['March 22, 1990', '(', '1990-03-22', ')'],
 'Running time': '95 minutes',
 'Country': 'United States',
 'Language': 'English'}

In [18]:
wiki_movies_raw[-5:]

[{'url': 'https://en.wikipedia.org/wiki/Holmes_%26_Watson',
  'year': 2018,
  'imdb_link': 'https://www.imdb.com/title/tt1255919/',
  'title': 'Holmes & Watson',
  'Directed by': 'Etan Cohen',
  'Produced by': ['Will Ferrell',
   'Adam McKay',
   'Jimmy Miller',
   'Clayton Townsend'],
  'Screenplay by': 'Etan Cohen',
  'Based on': ['Sherlock Holmes',
   'and',
   'Dr. Watson',
   'by',
   'Sir Arthur Conan Doyle'],
  'Starring': ['Will Ferrell',
   'John C. Reilly',
   'Rebecca Hall',
   'Rob Brydon',
   'Steve Coogan',
   'Ralph Fiennes'],
  'Music by': 'Mark Mothersbaugh',
  'Cinematography': 'Oliver Wood',
  'Edited by': 'Dean Zimmerman',
  'Productioncompanies ': ['Columbia Pictures',
   'Gary Sanchez Productions',
   'Mosaic Media Group',
   'Mimran Schur Pictures'],
  'Distributed by': 'Sony Pictures Releasing',
  'Release date': ['December 25, 2018',
   '(',
   '2018-12-25',
   ')',
   '(United States)'],
  'Running time': '90 minutes',
  'Country': 'United States',
  'Language

In [17]:
wiki_movies_raw[3600:3605]

[{'url': 'https://en.wikipedia.org/wiki/Benji:_Off_the_Leash!',
  'year': 2004,
  'imdb_link': 'https://www.imdb.com/title/tt0315273/',
  'title': 'Benji: Off the Leash!',
  'Directed by': 'Joe Camp',
  'Written by': 'Joe Camp',
  'Starring': ['Benji', 'Nick Whitaker', 'Shaggy', 'Gypsy the Cockatoo'],
  'Music by': 'Antonio di Lorenzo',
  'Productioncompany ': 'Mulberry Square Productions',
  'Distributed by': 'Mulberry Square Productions',
  'Release date': ['March 26, 2004', '(', '2004-03-26', ')'],
  'Running time': '97 min',
  'Country': 'United States',
  'Language': 'English',
  'Box office': '$3,817,362'},
 {'url': 'https://en.wikipedia.org/wiki/The_Best_Thief_in_the_World',
  'year': 2004,
  'imdb_link': 'https://www.imdb.com/title/tt0389796/',
  'title': 'The Best Thief in the World',
  'Directed by': 'Jacob Kornbluth',
  'Produced by': ['Tim Perrell', 'Nicola Usborne'],
  'Written by': 'Jacob Kornbluth',
  'Starring': ['Marc Rozendaal',
   'Michael Silverman',
   'David Warsh

At a raw glance the data appears to be organized

In [20]:
#Load MovieLens data obtained from Kaggle.com and examine datafames in subsequent cells
kaggle_metadata = pd.read_csv(f'{file_dir}movies_metadata.csv')
ratings = pd.read_csv(f'{file_dir}ratings.csv')

In [22]:
ratings.head()

userId  movieId  rating   timestamp
0       1      110     1.0  1425941529
1       1      147     4.5  1425942435
2       1      858     5.0  1425941523
3       1     1221     5.0  1425941546
4       1     1246     5.0  1425941556

In [23]:
kaggle_metadata.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

In [24]:
kaggle_metadata.columns

Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')

In [25]:
ratings.tail()

userId  movieId  rating   timestamp
26024284  270896    58559     5.0  1257031564
26024285  270896    60069     5.0  1257032032
26024286  270896    63082     4.5  1257031764
26024287  270896    64957     4.5  1257033990
26024288  270896    71878     2.0  1257031858

In [29]:
ratings.sample(10)

userId  movieId  rating   timestamp
24844067  258299     2124     3.0   974824100
9347986    96581     3039     5.0  1197074105
11088461  114760     3424     5.0  1181506442
25177784  261999    53322     3.5  1184608385
22355585  231992      107     3.0   860829485
24382385  253244     3478     4.0  1044465919
5732186    59033      480     4.5  1440296884
5998968    61749    77455     4.5  1300343972
11092698  114775     6448     1.5  1120062772
17929931  186095       47     4.5  1059694508

In [28]:
kaggle_metadata.tail()

adult belongs_to_collection budget  \
45461  False                   NaN      0   
45462  False                   NaN      0   
45463  False                   NaN      0   
45464  False                   NaN      0   
45465  False                   NaN      0   

                                                  genres  \
45461  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...   
45462                      [{'id': 18, 'name': 'Drama'}]   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   
45464                                                 []   
45465                                                 []   

                                   homepage      id    imdb_id  \
45461  http://www.imdb.com/title/tt6209470/  439050  tt6209470   
45462                                   NaN  111109  tt2028550   
45463                                   NaN   67758  tt0303758   
45464                                   NaN  227506  tt0008536   
45465                                   NaN  461257  tt6980792   

      original_language       original_title  \
45461                fa              رگ خواب   
45462                tl  Siglo ng Pagluluwal   
45463                en             Betrayal   
45464                en  Satana likuyushchiy   
45465                en             Queerama   

                                                overview  ... release_date  \
45461        Rising and falling between a man and woman.  ...          NaN   
45462  An artist struggles to finish his work while a...  ...   2011-11-17   
45463  When one of her hits goes wrong, a professiona...  ...   2003-08-01   
45464  In a small town live two brothers, one a minis...  ...   1917-10-21   
45465  50 years after decriminalisation of homosexual...  ...   2017-06-09   

      revenue runtime                          spoken_languages    status  \
45461     0.0    90.0    [{'iso_639_1': 'fa', 'name': 'فارسی'}]  Released   
45462     0.0   360.0         [{'iso_639_1': 'tl', 'name': ''}]  Released   
45463     0.0    90.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   
45464     0.0    87.0                                        []  Released   
45465     0.0    75.0  [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                          tagline                title  video  \
45461  Rising and falling between a man and woman               Subdue  False   
45462                                         NaN  Century of Birthing  False   
45463                      A deadly game of wits.             Betrayal  False   
45464                                         NaN     Satan Triumphant  False   
45465                                         NaN             Queerama  False   

      vote_average vote_count  
45461          4.0        1.0  
45462          9.0        3.0  
45463          3.8        6.0  
45464          0.0        0.0  
45465          0.0        0.0  

[5 rows x 24 columns]

Now comes the lengthy process of dealing with the wikipedia data and cleaning it up. This involved multiple cycles of filtering, cleaning, and renaming and reformatting. Each cycle involves inspecting the data, coming up with a plan of attack and then executing that plan.

In [30]:
#Here we convert the wikipedia json data to a df
wiki_movies_df = pd.DataFrame(wiki_movies_raw)

In [34]:
len(wiki_movies_df.columns)


193

In [35]:
list(wiki_movies_df.columns)

['url',
 'year',
 'imdb_link',
 'title',
 'Directed by',
 'Produced by',
 'Screenplay by',
 'Story by',
 'Based on',
 'Starring',
 'Narrated by',
 'Music by',
 'Cinematography',
 'Edited by',
 'Productioncompany ',
 'Distributed by',
 'Release date',
 'Running time',
 'Country',
 'Language',
 'Budget',
 'Box office',
 'Written by',
 'Genre',
 'Theme music composer',
 'Country of origin',
 'Original language(s)',
 'Producer(s)',
 'Editor(s)',
 'Production company(s)',
 'Original network',
 'Original release',
 'Productioncompanies ',
 'Executive producer(s)',
 'Production location(s)',
 'Distributor',
 'Picture format',
 'Audio format',
 'Voices of',
 'Followed by',
 'Composer(s)',
 'Created by',
 'Also known as',
 'Opening theme',
 'No. of episodes',
 'Preceded by',
 'Author',
 'Publisher',
 'Publication date',
 'Media type',
 'Pages',
 'ISBN',
 'OCLC',
 'LC Class',
 'Cover artist',
 'Series',
 'Set in',
 'Adaptation by',
 'Suggested by',
 'Biographical data',
 'Born',
 'Died',
 'Resti

A look at this dataframe reveals quite a huge mess to cleanup along with a large number of columns of which many are not needed for our analysis. Cleaning up dataframes such as this is a very lengthy process with multiple iterations and lots of trial and error

Looking at the list of columns, we see a host of data we do not need such as: 'Number of employees' or 'Divisision'. Lets start by removing refining our dataframe to only include column values that we need. In this case entries with an imdb_link and director 

In [44]:
wiki_movies_df[['Directed by', 'Director', 'imdb_link']].head(50)

Directed by Director  \
0                                        Renny Harlin      NaN   
1                                         James Foley      NaN   
2                                  Roger Spottiswoode      NaN   
3                                         Woody Allen      NaN   
4                                        John Cornell      NaN   
5                                         Larry Cohen      NaN   
6   [Barbara Kopple, Co-directors:, Cathy Caplan, ...      NaN   
7                                    Cedric Sundstrom      NaN   
8                                    Deborah Reinisch      NaN   
9                                         Eric Karson      NaN   
10                                        Walter Hill      NaN   
11                                     Frank Marshall      NaN   
12                                     Barry Levinson      NaN   
13                                     Penny Marshall      NaN   
14                                    Robert Zemeckis      NaN   
15                                      Curtis Hanson      NaN   
16                                  Frank Henenlotter      NaN   
17                                          Alan Alda      NaN   
18                                        John Badham      NaN   
19                                       Paul Wendkos      NaN   
20                                    Andy Blumenthal      NaN   
21                                    Kathryn Bigelow      NaN   
22                                     Brian De Palma      NaN   
23                                        Brian Yuzna      NaN   
24                                       Martin Sheen      NaN   
25                                    Roger Donaldson      NaN   
26                                     Roger Duchowny      NaN   
27                                        Albert Pyun      NaN   
28  [Milton Gray, Marsh Lamore, Bob Shellhorn, Mik...      NaN   
29              [Dennis Hopper, (as, Alan Smithee, )]      NaN   
30                                         Ben Jordan      NaN   
31                        [John Glen, Michael Levine]      NaN   
32                                          Mike Robe      NaN   
33                                         John Lafia      NaN   
34                                   James F. Collier      NaN   
35                                       Alan Metzger      NaN   
36                                      Robert Clouse      NaN   
37                                     Mark L. Lester      NaN   
38                                       Tony Wharmby      NaN   
39                                        Alan Parker      NaN   
40                                           Joe Roth      NaN   
41                                       Charles Band      NaN   
42          [Tony Bill, Barry L. Young (commercials)]      NaN   
43                                        John Waters      NaN   
44                                          Jack Fisk      NaN   
45                                      Kevin Costner      NaN   
46                                                NaN      NaN   
47                                       D.J. Webster      NaN   
48                                          Sam Raimi      NaN   
49                                      Stuart Gordon      NaN   

                                imdb_link  
0   https://www.imdb.com/title/tt0098987/  
1   https://www.imdb.com/title/tt0098994/  
2   https://www.imdb.com/title/tt0099005/  
3   https://www.imdb.com/title/tt0099012/  
4   https://www.imdb.com/title/tt0099018/  
5   https://www.imdb.com/title/tt0099026/  
6   https://www.imdb.com/title/tt0099028/  
7   https://www.imdb.com/title/tt0101326/  
8   https://www.imdb.com/title/tt0099037/  
9   https://www.imdb.com/title/tt0099039/  
10  https://www.imdb.com/title/tt0099044/  
11  https://www.imdb.com/title/tt0099052/  
12  https://www.imdb.com/title/tt0099073/  
13  https://www.imdb.com/title/tt0099077/  
14  https://www

In [45]:
#Lets generate a list movies that have a director and imdb link using list comprehension
wiki_movies = [movie for movie in wiki_movies_raw if('Director' in movie or 
                                                     'Directed by' in movie) and 
                                                    'imdb_link' in movie]
len(wiki_movies)

7080

The number of movies is slightly lower. If we convert the result to a dataframe we are left with 78 columns as some columns become empty as result

Again if we go back and look at columns we see that 'No. of episodes' is a column indicating that we are dealing with TV shows in some cases as opposed to movies

In [46]:
#Lets modify our list comprehension to account for this:
wiki_movies = [movie for movie in wiki_movies_raw if('Director' in movie or 'Directed by' in movie) 
                                                  and 'imdb_link' in movie 
                                                  and 'No. of episodes' not in movie]


In [47]:
len(wiki_movies)

7076

In [48]:
#Convert to dataframe
wiki_movies_df_new = pd.DataFrame(wiki_movies)

In [49]:
#View dataframe
wiki_movies_df_new.head(10)

url  year  \
0  https://en.wikipedia.org/wiki/The_Adventures_o...  1990   
1  https://en.wikipedia.org/wiki/After_Dark,_My_S...  1990   
2   https://en.wikipedia.org/wiki/Air_America_(film)  1990   
3    https://en.wikipedia.org/wiki/Alice_(1990_film)  1990   
4      https://en.wikipedia.org/wiki/Almost_an_Angel  1990   
5        https://en.wikipedia.org/wiki/The_Ambulance  1990   
6  https://en.wikipedia.org/wiki/American_Dream_(...  1990   
7  https://en.wikipedia.org/wiki/American_Ninja_4...  1990   
8     https://en.wikipedia.org/wiki/Andre%27s_Mother  1990   
9    https://en.wikipedia.org/wiki/Angel_Town_(film)  1990   

                               imdb_link                               title  \
0  https://www.imdb.com/title/tt0098987/     The Adventures of Ford Fairlane   
1  https://www.imdb.com/title/tt0098994/                After Dark, My Sweet   
2  https://www.imdb.com/title/tt0099005/                         Air America   
3  https://www.imdb.com/title/tt0099012/                               Alice   
4  https://www.imdb.com/title/tt0099018/                     Almost an Angel   
5  https://www.imdb.com/title/tt0099026/                       The Ambulance   
6  https://www.imdb.com/title/tt0099028/                      American Dream   
7  https://www.imdb.com/title/tt0101326/  American Ninja 4: The Annihilation   
8  https://www.imdb.com/title/tt0099037/                      Andre's Mother   
9  https://www.imdb.com/title/tt0099039/                          Angel Town   

                                         Directed by  \
0                                       Renny Harlin   
1                                        James Foley   
2                                 Roger Spottiswoode   
3                                        Woody Allen   
4                                       John Cornell   
5                                        Larry Cohen   
6  [Barbara Kopple, Co-directors:, Cathy Caplan, ...   
7                                   Cedric Sundstrom   
8                                   Deborah Reinisch   
9                                        Eric Karson   

                                     Produced by  \
0                     [Steve Perry, Joel Silver]   
1                    [Ric Kidney, Robert Redlin]   
2                                 Daniel Melnick   
3                                Robert Greenhut   
4                                   John Cornell   
5  [Larry Cohen, Moctesuma Esparza, Robert Katz]   
6                  [Arthur Cohn, Barbara Kopple]   
7                             Ovidio G Assonitis   
8                [Sarah Green, Deborah Reinisch]   
9                     [Ash R. Shah, Eric Karson]   

                                Screenplay by                     Story by  \
0  [David Arnott, James Cappe, Daniel Waters]  [David Arnott, James Cappe]   
1                [James Foley, Robert Redlin]                          NaN   
2                  [John Eskow, Richard Rush]                          NaN   
3                                         NaN                          NaN   
4                                         NaN                          NaN   
5                                         NaN                          NaN   
6                                         NaN                          NaN   
7                                         NaN                          NaN   
8                                         NaN                          NaN   
9                                         NaN                          NaN   

                                            Based on  \
0                        [Characters, by Rex Weiner]   
1  [the novel, After Dark, My Sweet, by, Jim Thom...   
2             [Air America, by, Christopher Robbins]   
3                                                NaN   
4                                                NaN   
5                                                NaN   
6                                           

Now we can use functions to clean our movie data:

In [ ]:
def clean_movie(movie):
    movie = dict(movie)  #Here the scope is limited and we can do what we want without affecting the global movie variable
    return movie

#We have essentially created a skeleton of function to work with our movies
#This will be expanded on below

In [51]:
#Note that the columns also contain values for languages such as Arabic or Chinese
#Let's investigate this further:
wiki_movies_df_new[wiki_movies_df_new['Arabic'].notnull()]

url  year  \
6834  https://en.wikipedia.org/wiki/The_Insult_(film)  2018   
7058   https://en.wikipedia.org/wiki/Capernaum_(film)  2018   

                                  imdb_link       title    Directed by  \
6834  https://www.imdb.com/title/tt7048622/  The Insult   Ziad Doueiri   
7058  https://www.imdb.com/title/tt8267604/   Capernaum  Nadine Labaki   

                                            Produced by  \
6834  [Rachid Bouchareb, Jean Bréhat, Julie Gayet, A...   
7058                    [Michel Merkt, Khaled Mouzanar]   

                                          Screenplay by  \
6834                                                NaN   
7058  [Nadine Labaki, Jihad Hojaily, Michelle Keserw...   

                                               Story by Based on  \
6834                                                NaN      NaN   
7058  [Georges Khabbaz, Nadine Labaki, Michelle Kese...      NaN   

                                               Starring  ... Hepburn  \
6834                       [Adel Karam, Kamel El Basha]  ...     NaN   
7058  [Zain Al Rafeea, Yordanos Shiferaw, Boluwatife...  ...     NaN   

        Literally Cantonese Chinese Yiddish       Arabic        Romanized  \
6834  Case No. 23       NaN     NaN     NaN  قضية رقم ٢٣  Qadiyya raqm 23   
7058          NaN       NaN     NaN     NaN    کفرناحوم‎              NaN   

     Russian Hebrew Polish  
6834     NaN    NaN    NaN  
7058     NaN    NaN    NaN  

[2 rows x 75 columns]

In the above we see two movies which we can get more info on based on url

In [52]:
wiki_movies_df_new[wiki_movies_df_new['Arabic'].notnull()]['url']

6834    https://en.wikipedia.org/wiki/The_Insult_(film)
7058     https://en.wikipedia.org/wiki/Capernaum_(film)
Name: url, dtype: object

A quick look shows that the 'Arabic' gives the movie's Arabic name. It doesn't have a name in any other language beside 'Arabic' and so those other language columns get NA values

Lets make a dictionary of alternate language titles for each movie. I.E: Arabic:" ".  This will take some investigation of column values.

In [54]:
#For example what exactly is 'Hepburn'
wiki_movies_df_new[wiki_movies_df_new['Hepburn'].notnull()]
#It would appear to be another language

url  year  \
6393  https://en.wikipedia.org/wiki/When_Marnie_Was_...  2015   
6829  https://en.wikipedia.org/wiki/Mary_and_the_Wit...  2018   
7009  https://en.wikipedia.org/wiki/My_Hero_Academia...  2018   
7054          https://en.wikipedia.org/wiki/Shoplifters  2018   

                                  imdb_link                         title  \
6393  https://www.imdb.com/title/tt3398268/         When Marnie Was There   
6829  https://www.imdb.com/title/tt6336356/   Mary and The Witch's Flower   
7009  https://www.imdb.com/title/tt7745068/  My Hero Academia: Two Heroes   
7054  https://www.imdb.com/title/tt8075192/                   Shoplifters   

               Directed by                                        Produced by  \
6393  Hiromasa Yonebayashi                                 Yoshiaki Nishimura   
6829  Hiromasa Yonebayashi                                 Yoshiaki Nishimura   
7009        Kenji Nagasaki  [Yoshihiro Oyabu, Koji Nagai, Wakana Okamura, ...   
7054     Hirokazu Kore-eda    [Matsuzaki Kaoru, Yose Akihiko, Taguchi Hijiri]   

                                         Screenplay by Story by  \
6393  [Masashi Andō, Keiko Niwa, Hiromasa Yonebayashi]      NaN   
6829            [Riko Sakaguchi, Hiromasa Yonebayashi]      NaN   
7009                                               NaN      NaN   
7054                                               NaN      NaN   

                                            Based on  \
6393   [When Marnie Was There, by, Joan G. Robinson]   
6829  [The Little Broomstick, by, Mary Stewart, [1]]   
7009         [My Hero Academia, by, Kōhei Horikoshi]   
7054                                             NaN   

                                               Starring  ...  \
6393                   [Sara Takatsuki, Kasumi Arimura]  ...   
6829       [Hana Sugisaki, Yūki Amami, Fumiyo Kohinata]  ...   
7009                                                NaN  ...   
7054  [Lily Franky, Sakura Ando, Mayu Matsuoka, Kair...  ...   

                                              Hepburn  \
6393                                   Omoide no Mānī   
6829                            Meari to Majo no Hana   
7009  Boku no Hīrō Akademia THE MOVIE: Futari no Hīrō   
7054                                   Manbiki Kazoku   

                    Literally Cantonese Chinese Yiddish Arabic Romanized  \
6393  Marnie of [my] Memories       NaN     NaN     NaN    NaN       NaN   
6829                      NaN       NaN     NaN     NaN    NaN       NaN   
7009                      NaN       NaN     NaN     NaN    NaN       NaN   
7054       Shoplifting Family       NaN     NaN     NaN    NaN       NaN   

     Russian Hebrew Polish  
6393     NaN    NaN    NaN  
6829     NaN    NaN    NaN  
7009     NaN    NaN    NaN  
7054     NaN    NaN    NaN  

[4 rows x 75 columns]

Unfortunately there is no shortcut to this. We must go through each column to determine which contain alternate titles for each film

In [63]:
col_values = (list(wiki_movies_df_new.columns))


In [68]:
col_values.sort(reverse=False)

In [69]:
col_values

['Adaptation by',
 'Also known as',
 'Animation by',
 'Arabic',
 'Audio format',
 'Based on',
 'Box office',
 'Budget',
 'Cantonese',
 'Chinese',
 'Cinematography',
 'Color process',
 'Composer(s)',
 'Country',
 'Country of origin',
 'Created by',
 'Directed by',
 'Director',
 'Distributed by',
 'Distributor',
 'Edited by',
 'Editor(s)',
 'Executive producer(s)',
 'Followed by',
 'French',
 'Genre',
 'Hangul',
 'Hebrew',
 'Hepburn',
 'Japanese',
 'Label',
 'Language',
 'Length',
 'Literally',
 'Mandarin',
 'McCune–Reischauer',
 'Music by',
 'Narrated by',
 'Original language(s)',
 'Original network',
 'Original release',
 'Original title',
 'Picture format',
 'Polish',
 'Preceded by',
 'Produced by',
 'Producer',
 'Producer(s)',
 'Production company(s)',
 'Production location(s)',
 'Productioncompanies ',
 'Productioncompany ',
 'Recorded',
 'Release date',
 'Released',
 'Revised Romanization',
 'Romanized',
 'Running time',
 'Russian',
 'Screen story by',
 'Screenplay by',
 'Simplifie

In [71]:
wiki_movies_df_new.iloc[2]['title']

'Air America'

In [85]:
for i in range(len(col_values)):
    name = col_values[i]
    print(str(col_values[i]) + ": " + str(wiki_movies_df_new.iloc[3][name]))
    

Adaptation by: nan
Also known as: nan
Animation by: nan
Arabic: nan
Audio format: nan
Based on: nan
Box office: $7,331,647
Budget: $12 million
Cantonese: nan
Chinese: nan
Cinematography: Carlo Di Palma
Color process: nan
Composer(s): nan
Country: United States
Country of origin: nan
Created by: nan
Directed by: Woody Allen
Director: nan
Distributed by: Orion Pictures
Distributor: nan
Edited by: Susan E. Morse
Editor(s): nan
Executive producer(s): nan
Followed by: nan
French: nan
Genre: nan
Hangul: nan
Hebrew: nan
Hepburn: nan
Japanese: nan
Label: nan
Language: English
Length: nan
Literally: nan
Mandarin: nan
McCune–Reischauer: nan
Music by: nan
Narrated by: nan
Original language(s): nan
Original network: nan
Original release: nan
Original title: nan
Picture format: nan
Polish: nan
Preceded by: nan
Produced by: Robert Greenhut
Producer: nan
Producer(s): nan
Production company(s): nan
Production location(s): nan
Productioncompanies : nan
Productioncompany : nan
Recorded: nan
Release date

In [91]:
wiki_movies_df_new[wiki_movies_df_new['Label'].notnull()][['Label','title']]

Label  \
963   [Warner Music Vision, Warner-Reprise Video, Ma...   
1347                                           Columbia   

                                 title  
963   The Girlie Show: Live Down Under  
1347                    The Nona Tapes

In [93]:
wiki_movies_df_new[wiki_movies_df_new['Polish'].notnull()][['Polish','title']]

Polish     title
7070  Zimna wojna  Cold War

In [94]:
wiki_movies_df_new.head(1)

url  year  \
0  https://en.wikipedia.org/wiki/The_Adventures_o...  1990   

                               imdb_link                            title  \
0  https://www.imdb.com/title/tt0098987/  The Adventures of Ford Fairlane   

    Directed by                 Produced by  \
0  Renny Harlin  [Steve Perry, Joel Silver]   

                                Screenplay by                     Story by  \
0  [David Arnott, James Cappe, Daniel Waters]  [David Arnott, James Cappe]   

                      Based on  \
0  [Characters, by Rex Weiner]   

                                            Starring  ... Hepburn Literally  \
0  [Andrew Dice Clay, Wayne Newton, Priscilla Pre...  ...     NaN       NaN   

  Cantonese Chinese Yiddish Arabic Romanized Russian Hebrew Polish  
0       NaN     NaN     NaN    NaN       NaN     NaN    NaN    NaN  

[1 rows x 75 columns]

In [99]:
for i in range(len(col_values)):
    name = col_values[i]
    df = wiki_movies_df_new[wiki_movies_df_new[name].notnull()][['title', name]]
    print(df.head(1)['title'] + ": ")
    print (df.head(1)[name])
    print ("\n")    

534    The Last of the Mohicans: 
Name: title, dtype: object
534    [John L. Balderston, Paul Perez, Daniel Moore]
Name: Adaptation by, dtype: object


1342    Nightwatch: 
Name: title, dtype: object
1342    Detonator II: Night Watch
Name: Also known as, dtype: object


1377    Runaway Brain: 
Name: title, dtype: object
1377    [Andreas Deja, Gary Dunn, Deboissy Sylvain]
Name: Animation by, dtype: object


6834    The Insult: 
Name: title, dtype: object
6834    قضية رقم ٢٣
Name: Arabic, dtype: object


26    Camp Cucamonga: 
Name: title, dtype: object
26    Stereo
Name: Audio format, dtype: object


0    The Adventures of Ford Fairlane: 
Name: title, dtype: object
0    [Characters, by Rex Weiner]
Name: Based on, dtype: object


0    The Adventures of Ford Fairlane: 
Name: title, dtype: object
0    $21.4 million
Name: Box office, dtype: object


0    The Adventures of Ford Fairlane: 
Name: title, dtype: object
0    $20 million
Name: Budget, dtype: object


6456    Ip Man 3: 
Name: title

If we look at our list we see that there are several columns that appear to contain alternate titles: Also known as,
                  Arabic,
                  Cantonese,
                  Chinese,
                  French,
                  Hangul,
                  Hebrew,
                  Hepburn,
                  Japanese,
                  Literally,
                  Mandarin,
                  McCune-Reischauer,
                  Original Title,
                  Polish,
                  Revised Rominization,
                  Romanized,
                  Russian,
                  Simplified,
                  Traditional,
                  Yiddish
                  

In [107]:
wiki_movies[0]

{'url': 'https://en.wikipedia.org/wiki/The_Adventures_of_Ford_Fairlane',
 'year': 1990,
 'imdb_link': 'https://www.imdb.com/title/tt0098987/',
 'title': 'The Adventures of Ford Fairlane',
 'Directed by': 'Renny Harlin',
 'Produced by': ['Steve Perry', 'Joel Silver'],
 'Screenplay by': ['David Arnott', 'James Cappe', 'Daniel Waters'],
 'Story by': ['David Arnott', 'James Cappe'],
 'Based on': ['Characters', 'by Rex Weiner'],
 'Starring': ['Andrew Dice Clay',
  'Wayne Newton',
  'Priscilla Presley',
  'Lauren Holly',
  'Morris Day',
  'Robert Englund',
  "Ed O'Neill"],
 'Narrated by': 'Andrew "Dice" Clay',
 'Music by': ['Cliff Eidelman', 'Yello'],
 'Cinematography': 'Oliver Wood',
 'Edited by': 'Michael Tronick',
 'Productioncompany ': 'Silver Pictures',
 'Distributed by': '20th Century Fox',
 'Release date': ['July 11, 1990', '(', '1990-07-11', ')'],
 'Running time': '102 minutes',
 'Country': 'United States',
 'Language': 'English',
 'Budget': '$20 million',
 'Box office': '$21.4 milli

In [130]:
#Now we need to make a function to handle alternate titles:

def clean_movie(movie):
#Make an empty dictionary to hold all alternative titles:
    alt_titles = {}
    alt_names  = ['Also known as', 
                 'Arabic', 
                 'Cantonese', 
                 'Chinese', 
                 'French', 
                 'Hangul', 
                 'Hebrew', 
                 'Hepburn', 
                 'Japanese', 
                 'Literally', 
                 'Mandarin', 
                 'McCune-Reischauer', 
                 'Original Title', 
                 'Polish', 
                 'Revised Rominization', 
                 'Romanized', 
                 'Russian', 
                 'Simplified', 
                 'Traditional', 
                 'Yiddish']
    movie = dict(movie)
#Loop through a list of alternative title keys
    for alt_name in alt_names:
        #check if key exists in movie object
        if alt_name in movie:
#If so remove key value pair and add to alt_titles dictionary
            alt_titles[alt_name] = movie[alt_name]
            movie.pop(alt_name)
#Aftr looping through every key add the alternative title dictionary to movie object
    if len(alt_titles) > 0:
        movie['alt_titles'] = alt_titles  #Note how the whole dict is now stored in alt titles
    return movie

In [131]:
#Now that we have our function, we can use it to make a list of clean movies:
clean_movies = [clean_movie(movie) for movie in wiki_movies]

In [136]:
#Lets make this into a dataframe:
wiki_movies_clean_df = pd.DataFrame(clean_movies)

In [140]:
#Now lets see the list of columns
#Note that alt_titles has been added:

clean_columns = list(wiki_movies_clean_df.columns)
clean_columns.sort(reverse = False)
clean_columns

['Adaptation by',
 'Animation by',
 'Audio format',
 'Based on',
 'Box office',
 'Budget',
 'Cinematography',
 'Color process',
 'Composer(s)',
 'Country',
 'Country of origin',
 'Created by',
 'Directed by',
 'Director',
 'Distributed by',
 'Distributor',
 'Edited by',
 'Editor(s)',
 'Executive producer(s)',
 'Followed by',
 'Genre',
 'Label',
 'Language',
 'Length',
 'McCune–Reischauer',
 'Music by',
 'Narrated by',
 'Original language(s)',
 'Original network',
 'Original release',
 'Original title',
 'Picture format',
 'Preceded by',
 'Produced by',
 'Producer',
 'Producer(s)',
 'Production company(s)',
 'Production location(s)',
 'Productioncompanies ',
 'Productioncompany ',
 'Recorded',
 'Release date',
 'Released',
 'Revised Romanization',
 'Running time',
 'Screen story by',
 'Screenplay by',
 'Starring',
 'Story by',
 'Suggested by',
 'Theme music composer',
 'Venue',
 'Voices of',
 'Written by',
 'alt_titles',
 'imdb_link',
 'title',
 'url',
 'year']